### ISSUE TITLE
Respect SemVer

### ISSUE THREAD

**Issue by tristanbes** [shentao/vue-multiselect#767]
Hello,

Please, consider using [semver](https://semver.org/)
The v2.1.0 should be a v3.0.0 since there is a BC break.

I spent hour(s) to trying to understand what happened/debug, which is also my bad because I thought everyone is respecting this standard when it comes to updating libraries :s

**Comment by shentao** [comment]
Hey, that is very sad to hear that. Could you please explain what exactly was the breaking change? 

The biggest change in v2.1.0 was the removal of the internal value (which was deep-cloned from the value), but that was a private change which did not affect the API. I imagine it might have affected some projects that abused the fact that there was a local state, but that wasn’t a feature to begin with, just implementation detail. The API nor the feature set has not been changed.

**Comment by tristanbes** [comment]
Hello @shentao , the project depended on the internal value to answer your question. I was not the author of the code so I can't give you more insights, but even if it's internal, it should be considered as a BC break

**Comment by shentao** [comment]
Major updates in SemVer happen when you make backward incompatible changes to the public API. I just googled that just to make sure. This is not something that happened in this case.

I did remove the `internalValue` which was a deep-cloned version of the `value` that was updated whenever some actions happened inside the component. Whenever a change happened, it emitted the internalValue for the user to update their outside value. Sadly this has been very problematic since it was prone to breaking some reactivity rules. It was changed so that the component now fully derives the internalValue and never mutates anything. That’s why I imagine the code in your project might have been broken by the change. It must have been dependant on Multiselect updating its own internalValue even though that’s the developer responsibility and the docs state that you always want to have either `v-model` or `@input` to handle `value` updates on your end.

I understand your frustration and I’m sorry to hear that. I hope this won’t happen again as I will try to not make any stupid design decisions as I first did when introducing the internalValue (a few years ago in v1).
